File: pySDK/CONST.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
USR_GPIO_CNT = 8
USR_BTN_CNT = 6
RcPWMchanNum = 6

GPIOgroupCntStepMotor = 2

FC_CTRL_BLE_MSGTYPE_LEN = 2
etRoll_PwmCH = 0
etPitchPwmCH = 1
etThrotPwmCH = 2
etRudd_PwmCH = 3
etAUX1_PwmCH = 4
etAUX2_PwmCH = 5
etFlightModeCH = 6
etAUX4_CH = 7
etAUX5_CH = 8
etAUX6_CH = 9
etAUX7_CH = 10
etGPIO_CTRL_BYTE = 11

CHAR3_BYTE_LEN = 20
StepMotorCnt = 2

FC_STATUS_LEN = 20
FC_STATUS_EACH_AXIS_BYTE = 5
RPY_floatIntConversion = 2
LSB_DROP_ACC = 5
ACC_DYNAMIC_FS_RNG = 2.0 # // g. MPU6050_ACCEL_FS_2: \pm 2G
LSB_DROP_GYR = 0
GYRO_DYNAMIC_FS_RNG = 2000.0 # // deg/sec
LSB_DROP_MAG = 0
LSBcountPerGauss = 1024 # earth field: 0.25 to 0.65 gauss
ByteIndex0Seq_n_Btn = 15
ByteIndex0ADC1_1tenthV = 16
ByteIndex0Compass = 17
ByteIndex0Tmpture = 18


MinMsIntervalForInternetCtrl = 500 # if py2app cmd update rate is faster than this duration, app may experience packet loss 
readApp2pyDataInterval_ms = 500
strApp2pyCmdFilename = "app2pyCmd"  
strPy2appJsonCmdFilename = "py2appJsonCmd"  

etAndroid = "android"
etDebian = "Debian"
etInternet = "mqtt"
etLocalNet = "LocalNet"

platformTypes = [etAndroid,etDebian,etInternet,etLocalNet]

TOPIC_APP_INTERNET_MQTT_CTRL = "INTERNET_CTRL"
TOPIC_APP_INTERNET_MQTT_MetaCtrl = "INTERNET_META_CTRL"
strREQUEST_Telemetry = "requestTelemetry"; # in order to save internet traffic, we may turn off the Char4 notification, and request it as needed
TOPIC_APP_INTERNET_Char4_TELEMETRY = "INTERNET_TELEMETRY"
MinMsIntervalApp2SDKnotification = 500;

stepMotorRollIndex = 0
stepMotorPitchIndex = 1

socketServerPORT_pyPrintWindow = 28090
socketServerPORT_SDKpy2app = 28091 # app as server
socketServerPORT_SDKapp2py = 28092 # app as client

spiderPWMchSpeed = 1
spiderPWMchSteer = 0

File: javaSDK/CONST.java
public class CONST {
	public static final int FC_STATUS_EACH_AXIS_BYTE = 5;
	public static final int RPY_floatIntConversion = 2;
	public static final int LSB_DROP_ACC = 5;
	public static final float ACC_DYNAMIC_FS_RNG = 2.0f; // g. MPU6050_ACCEL_FS_2: \pm 2G
	public static final float GYRO_DYNAMIC_FS_RNG = 2000.0f; // deg/sec
	public static final int LSBcountPerGauss = 6842; // LIS3MDL, \pm 4Gauss full range
	public static final int ByteIndex0Compass = 17;
	public static final int LSB_DROP_MAG = 6;
	public static final int LSB_DROP_GYR = 0;
	public static final int ByteIndex0Tmpture = 18;
	public static final int CHAR3_BYTE_LEN = 20;
    public static final int FC_CTRL_BLE_MSGTYPE_LEN = 2;
    public static final byte FC_CTRL_MSG_PREAMBLE_TYPE_TX[]      = new byte[] {(byte)0x0, (byte)0xFF};
    public static final byte FC_CTRL_MSG_PREAMBLE_TYPE_SETNAME[] = new byte[] {(byte)0x55, (byte)0xAA};
    public static final byte FC_CTRL_MSG_PREAMBLE_TYPE_SET_PID[] = new byte[] {(byte)0x17, (byte)0x88};
    public static final int FC_CTRL_MSG_SET_PID_Byte0_forValue = 4;
    public static final int RcPWMchanNum = 6;
    public static final int HbridgeCnt = RcPWMchanNum/2;
    static final byte SeqIDbitCnt = 2;
    static final byte etRoll_PwmCH = 0;
    static final byte etPitchPwmCH = 1;static final byte RcJoystickCHnum = 2;
    static final byte etThrotPwmCH = 2;
    static final byte etRudd_PwmCH = 3;
    static final byte etAUX1_PwmCH = 4;
    static final byte etAUX2_PwmCH = 5;
    static final byte etFlightModeSeqIdCH = 6; // realtime settings, flight mode. MSB=>LSB, AUX3_FightModeMask, SeqIDbitCnt
    static final byte etAUX4_CH = 7;
    static final byte etAUX5_CH = 8;
    static final byte etAUX6_CH = 9;
    static final byte etAUX7_CH = 10;
    static final byte etGPIO_CTRL_BYTE = 11;
    static final byte RcAllCHnum = 12;
    public static final int StepMotorCnt = 2;
    public static final int stepMotorRollIndex = 0;
    public static final int stepMotorPitchIndex = 1;
    public enum fightModeSetting{
        DirectCmdMode("DirectCmdMode",(byte)0),
        AttitudeCtrlMode("AttitudeCtrlMode",(byte)1);
        private String m_strname;
        private byte m_byteEnumVal;
        private fightModeSetting(String stringname, byte byteEnumVal) {
            this.m_strname = stringname;
            this.m_byteEnumVal = byteEnumVal;
        }
        @Override
        public String toString(){
            return m_strname;
        }
        public byte get_byteEnumVal() { return m_byteEnumVal; }
    }
    
    
    
}
